File: DP(Dynamic_Programming, 동적계획법)/백준_1699_제곱수의 합/제곱수의 합.java
import java.util.*;

public class Main {
    static int[] arr = new int[100000];

    public static void main(String[] args) {

        Scanner s = new Scanner(System.in);
        int input = s.nextInt();

        for(int i = 0; i<= input; i++){
            arr[i] = i;
            for(int j = 0; j*j <= i; j++){
                arr[i] = Math.min(arr[i], arr[i-j*j] + 1);
            }
        }

        System.out.println(arr[input]);
    }

}

File: DP(Dynamic_Programming, 동적계획법)/백준_1699_제곱수의 합/제곱수의 합.kt
import java.util.*
import kotlin.math.*

fun main(args: Array<String>){
    val arr = IntArray(100000)

    val s = Scanner(System.`in`)
    val input = s.nextInt()

    for(i in 0..input){
        arr[i] = i
        var j = 0
        while(j*j<=i){
            arr[i] = min(arr[i], arr[i-j*j] + 1)
            j++
        }
    }
    print(arr[input])

}